Adding the multimodal RAG tutorial with Amazon Nova and LangChain

debnsuma opened the pull request:
This notebook demonstrates how to implement a multi-modal Retrieval-Augmented Generation (RAG) system using Amazon Bedrock with Amazon Nova and LangChain. Many documents contain a mixture of content types, including text and images. Traditional RAG applications often lose valuable information captured in images. With the emergence of Multimodal Large Language Models (MLLMs), we can now leverage both text and image data in our RAG systems.

3coins commented on samples/multi-modal/multimodal_rag_with_nova.ipynb:
```diff
@@ -0,0 +1,767 @@
+    "display.Markdown(response)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h2 style=\"background: linear-gradient(to right, #ff6b6b, #4ecdc4, #1e90ff); \n",
+    "            color: white; \n",
+    "            padding: 15px; \n",
+    "            border-radius: 10px; \n",
+    "            text-align: center; \n",
+    "            font-family: 'Comic Sans MS', cursive, sans-serif; \n",
+    "            text-shadow: 2px 2px 4px rgba(0,0,0,0.5);\">\n",
+    "  Thank you!\n",
+    "</h2>"
+   ]
+  }
+ ],
```
Replace with a markdown header.

3coins reviewed (commented):
@debnsuma
This is a a great addition, thanks for submitting this. Added a few suggestions to simplify some of the code and structure.

debnsuma commented:
Thanks so much @3coins for all your inputs. I fixed all of them and refactored the notebook and pushed the changes. 
Just one feedback, I am still exploring on how to perform the embeddings using BedrockEmbeddings, I tried before, but I was not able to do that to generate embeddings for both image and text as Titan's query request schema is different for image and text.
